src/main/java/teammates/common/datatransfer/attributes/FeedbackSessionAttributes.java

```diff
@@ -43,100 +46,76 @@
+
+        instructions = new Text("");
+    }
+
+    public static FeedbackSessionAttributes valueOf(FeedbackSession fs) {
+        return builder(fs.getFeedbackSessionName(), fs.getCourseId(), fs.getCreatorEmail())
+                .withInstructions(fs.getInstructions())
+                .withCreatedTime(fs.getCreatedTime())
+                .withStartTime(fs.getStartTime())
+                .withEndTime(fs.getEndTime())
+                .withSessionVisibleFromTime(fs.getSessionVisibleFromTime())
+                .withResultsVisibleFromTime(fs.getResultsVisibleFromTime())
+                .withTimeZone(fs.getTimeZone())
+                .withGracePeriod(fs.getGracePeriod())
+                .withFeedbackSessionType(fs.getFeedbackSessionType())
+                .withSentOpenEmail(fs.isSentOpenEmail())
+                .withSentClosingEmail(fs.isSentClosingEmail())
+                .withSentClosedEmail(fs.isSentClosedEmail())
+                .withSentPublishedEmail(fs.isSentPublishedEmail())
+                .withOpeningEmailEnabled(fs.isOpeningEmailEnabled())
+                .withPublishedEmailEnabled(fs.isPublishedEmailEnabled())
+                .withRespondingInstructorList(fs.getRespondingInstructorList())
+                .withRespondingStudentList(fs.getRespondingStudentList())
+                .withClosingEmailEnabled(fs.isClosingEmailEnabled())
+                .build();
     }
 
-    private FeedbackSessionAttributes(FeedbackSessionAttributes other) {
-        this(other.feedbackSessionName, other.courseId, other.creatorEmail,
-            other.instructions, other.createdTime, other.startTime, other.endTime,
-            other.sessionVisibleFromTime, other.resultsVisibleFromTime, other.timeZone,
-            other.gracePeriod, other.feedbackSessionType,
-            other.sentOpenEmail, other.sentClosingEmail, other.sentClosedEmail, other.sentPublishedEmail,
-            other.isOpeningEmailEnabled, other.isClosingEmailEnabled,
-            other.isPublishedEmailEnabled, other.respondingInstructorList,
-            other.respondingStudentList);
+    /**
+     * Return new builder instance with default values for optional fields.
+     *
+     * <p>Following default values are set to corresponding attributes:
+     * <ul>
+     * <li>{@code isOpeningEmailEnabled = true}</li>
+     * <li>{@code isClosingEmailEnabled = true}</li>
+     * <li>{@code isPublishedEmailEnabled = true}</li>
+     * <li>{@code respondingInstructorList = new HashSet<>()}</li>
+     * <li>{@code respondingStudentList = new HashSet<>()}</li>
+     * </ul>
+     */
+    public static Builder builder(String feedbackSessionName, String courseId, String creatorEmail) {
+        return new Builder(feedbackSessionName, courseId, creatorEmail);
     }
 
     public FeedbackSessionAttributes getCopy() {
-        return new FeedbackSessionAttributes(this);
+        return builder(feedbackSessionName, courseId, creatorEmail)
```

Comment:
Will `this(toEntity())` (essentially convert to entity and back to attributes) work? If it works, it significantly reduces repetition.